apps/backend/src/modules/courses/course.controller.ts

```diff
@@ -18,15 +53,229 @@ class CourseController extends BaseController {
+					options as APIHandlerOptions<{
+						body: AddCourseRequestDto;
+					}>,
+				),
+			method: "POST",
+			path: CoursesApiPath.ROOT,
+			validation: {
+				body: addCourseValidationSchema,
+			},
+		});
+		this.addRoute({
+			handler: (options) =>
+				this.delete(
+					options as APIHandlerOptions<{
+						params: { courseId: string };
+					}>,
+				),
+			method: "DELETE",
+			path: CoursesApiPath.$COURSE_ID,
+			validation: {
+				params: courseIdParameterValidationSchema,
+			},
+		});
+		this.addRoute({
+			handler: (options) =>
+				this.find(
+					options as APIHandlerOptions<{
+						params: { courseId: string };
+					}>,
+				),
+			method: "GET",
+			path: CoursesApiPath.$COURSE_ID,
+			validation: {
+				params: courseIdParameterValidationSchema,
+			},
+		});
+		this.addRoute({
+			handler: (options) =>
+				this.findAllByVendors(
+					options as APIHandlerOptions<{
+						query: CourseSearchRequestDto;
+						user: UserAuthResponseDto;
+					}>,
+				),
 			method: "GET",
 			path: CoursesApiPath.ROOT,
 		});
+		this.addRoute({
+			handler: (options) =>
+				this.update(
+					options as APIHandlerOptions<{
+						params: { courseId: string };
+					}>,
+				),
+			method: "PUT",
+			path: CoursesApiPath.$COURSE_ID,
+			validation: {
+				params: courseIdParameterValidationSchema,
+			},
+		});
+	}
+
+	/**
+	 * @swagger
+	 * /courses:
+	 *    post:
+	 *      description: Fetch course from vendor API and add for user in DB
+	 *      requestBody:
+	 *        required: true
+	 *        content:
+	 *          application/json:
+	 *            schema:
+	 *              type: object
+	 *              properties:
+	 *                vendorCourseId:
+	 *                  type: string
+	 *                vendorId:
+	 *                  type: number
+	 *                  format: number
+	 *                  minimum: 1
+	 *      responses:
+	 *        200:
+	 *          description: Successful operation
+	 *          content:
+	 *            application/json:
+	 *              schema:
+	 *                type: object
+	 *                $ref: "#/components/schemas/Course"
+	 */
+	private async create({
+		body,
+	}: APIHandlerOptions<{
+		body: AddCourseRequestDto;
+	}>): Promise<APIHandlerResponse> {
+		return {
+			payload: await this.courseService.create(body),
+			status: HTTPCode.CREATED,
+		};
+	}
+
+	/**
+	 * @swagger
+	 * /courses:
+	 *    delete:
+	 *      description: Delete course by id
+	 *      responses:
+	 *        200:
+	 *          description: Successful operation
+	 *          content:
+	 *            application/json:
+	 *              schema:
+	 *                type: object
+	 *                properties:
+	 *                  success:
+	 *                    type: boolean
+	 */
+	private async delete({
+		params: { courseId },
+	}: APIHandlerOptions<{
+		params: { courseId: string };
+	}>): Promise<APIHandlerResponse> {
+		const success = await this.courseService.delete(Number.parseInt(courseId));
```

Comment:
```suggestion
		const success = await this.courseService.delete(Number(courseId));
```
You can check in validation if it is an integer or no